aspnetcore/performance/caching/hybrid.md

```diff
@@ -0,0 +1,179 @@
+
+* MaximumPayloadBytes - Maximum size of a cache entry. Default value is 1 MB. Attempts to store values over this size are logged, and the value is not stored in cache.
+* MaximumKeyLength - Maximum length of a cache key. Default value is 1024 characters. Attempts to store values over this size are logged, and the value is not stored in cache.
+ 
+## Serialization
+
+Serialization is configured as part of registering the service, with support for type-specific and generalized serializers via the
+`WithSerializer` and `.WithSerializerFactory` methods, chained from the `AddHybridCache` call. By default, the library
+handles `string` and `byte[]` internally, and uses `System.Text.Json` for everything else, but you can configure `HybridCache` to use protobuf, xml, or anything else.
+
+An example that configures the service to use a protobuf serializer is [here](https://github.com/dotnet/aspnetcore/commit/ac0dbd8c96caa704e39a080b8db59183b6dd3e13).
+
+## Cache storage
+
+By default `HybridCache` uses <xref:System.Runtime.Caching.MemoryCache> for its primary cache storage. Cache entries are stored in-process, so each server has a separate cache that is lost whenever the server process is restarted. For secondary out-of-process storage, such as Redis or SQL Server, `HybridCache` uses [the configured `IDistributedCache` implementation](xref:performance/caching/distributed), if any. But even without an `IDistributedCache`implementation, the `HybridCache` service still provides in-process caching and stampede protection.
+
+## Optimize performance
+
+To optimize performance, configure `HybridCache` to reuse objects and avoid `byte[]` allocations.
+
+### Reuse objects
+
+In typical existing code that uses `IDistributedCache`, every retrieval of an object from the cache results in deserialization. This behavior means that each concurrent caller gets a separate instance of the object, which cannot interact with other instances. The result is thread safety, as there's no risk of concurrent modifications to the same object instance.
+
+Because a lot of `HybridCache` usage will be adapted from existing `IDistributedCache` code, `HybridCache` preserves this behavior by default to avoid introducing concurrency bugs. However, objects are inherently thread-safe if:
+
+* They are immutable types.
+* The code doesn't modify them.
+
+In such cases, inform `HybridCache` that it's safe to reuse instances by:
+
+* Marking the type as `sealed`. The `sealed` keyword in C# means that the class cannot be inherited.
+* Applying the `[ImmutableObject(true)]` attribute to the type. The `[ImmutableObject(true)]` attribute indicates that the object's state cannot be changed after it's created.
+
+By reusing instances, `HybridCache` can reduce the overhead of CPU and object allocations associated with per-call deserialization. This can lead to performance improvements in scenarios where the cached objects are large or accessed frequently.
```

Comment:
Very minor suggestion, not required: Consider moving this paragraph to the intro for this section.  I thought it would be helpful to understand this for some context before we jump into the details.